openssl-dynamic/src/main/c/ssl.c

```diff
@@ -721,10 +781,24 @@ TCN_IMPLEMENT_CALL(jint, SSL, initialize)(TCN_STDARGS, jstring engine)
                 if (!ENGINE_set_default(ee, ENGINE_METHOD_ALL))
                     err = APR_ENOTIMPL;
             }
-            /* Free our "structural" reference. */
-            if (ee)
+            if (ee != NULL) {
```

Comment:
The error flow of this is still hard to follow, because it is possible for `err == APR_ENOTIMPL` here. It seems like that isn't on purpose. Although it looks like it would work. Is that on purpose?

Reply:
yeah let me just make it more clear.